### ISSUE TITLE
Move disableInlineStyleAttributesParsing to HtmlPruner?

### ISSUE THREAD

**Issue by oliverklee** [MyIntervals/emogrifier#775]
@zoliszabo, @JakeQZ, I propose we deprecated this option and instead add a method `removeAllStyleAttributes` or `removeInlineStyles` to the `HtmlPruner` class (which then can be used before `CssInliner` gets used). What do you think?

**Comment by JakeQZ** [comment]
Makes sense - as they can share the same `DOMDocument`, and there's no reason (I can think of) why `HtmlPruner` can't be instantiated first, then `CssInliner` with `::fromDomDocument`, and the original `HtmlPruner` can still be used after `inlineCss` for post-processing.

**Comment by JakeQZ** [comment]
It might be an idea to initially mark the method as deprecated for 4.0, and change the current implementation to create a temporary `HtmlPruner` to invoke the functionality (though a quick look at the code indicates that won't be entirely straightforward, due to the way `style` attributes are initially removed anyway, stored in `styleAttributesForNodes` and then later merged with the inlined styles - but it should be possible to introduce an initial step in `inlineCss` to remove `style` attributes if the option is enabled [default], and have the remaining code not worry about that option).

**Comment by oliverklee** [comment]
> It might be an idea to initially mark the method as deprecated for 4.0

Yes, that's what I was thinking of.

As for the refactoring by using `HtmlPruner` internally, I'd like to avoid this as this code will be removed anyway.

**Comment by JakeQZ** [comment]
> As for the refactoring by using `HtmlPruner` internally, I'd like to avoid this as this code will be removed anyway.

That was partly based on a presumption that there'd be piece of code specifically and only removing `style` attributes, which on looking more closely turns out not to be the case.  But it would still be possible to separate out the removal of `style` attributes, so that the remaining code becomes as it will be when the feature is finally removed from the `CssInliner` class, if you wanted to make the changes to the remaining code sooner rather than later.

**Comment by oliverklee** [comment]
> But it would still be possible to separate out the removal of `style` attributes, so that the remaining code becomes as it will be when the feature is finally removed from the `CssInliner` class

Oh, yes, that would be excellent.
